optimizedJitJarAll ignores shadowJar zip64 config, breaking the build

### Expected Behavior

I'm trying to build our API, but we're now over the 32 bit limit, so I have to enable zip64. However, it looks like micronaut or shadowJar is ignoring the zip64 config option when running the command `./gradlew optimizedJitJarAll`. Even after adding `shadowJar { zip64 true }`, I still get the error `org.apache.tools.zip.Zip64RequiredException: archive contains more than 65535 entries.` Is there anything I'm doing wrong here? This is a showstopper for us, any help is appreciated.

Excerpt from my build.gradle:
```groovy
plugins {
    id 'org.jetbrains.kotlin.jvm' version "$kotlinVersion"
    id 'org.jetbrains.kotlin.plugin.allopen' version "$kotlinVersion"
    id 'com.google.devtools.ksp' version "$kspVersion"

    // shadow plugin is used to build a "fat jar" that is required when building the Dockerfile. 
    id "com.github.johnrengelman.shadow" version "8.1.1"

    id "io.micronaut.minimal.application" version "$micronautPlatformVersion"
    id "io.micronaut.aot" version "$micronautPlatformVersion"
    id "io.micronaut.test-resources" version "$micronautPlatformVersion"
    id 'com.jfrog.artifactory' version '5.1.6'
    id "jacoco"
    id "org.sonarqube" version "4.3.1.3277"
    id "org.flywaydb.flyway" version "9.22.1"
}

micronaut {
    version "$micronautVersion" // defined in gradle.properties
    runtime "netty"
    testRuntime "junit5"
    processing {
        incremental true
        annotations "com.cfa.ots.fsd.*"
    }

    aot {
        // Please review carefully the optimizations enabled below
        // Check https://micronaut-projects.github.io/micronaut-aot/latest/guide/ for more details
        optimizeServiceLoading = false
        convertYamlToJava = false
        precomputeOperations = true
        cacheEnvironment = true
        optimizeClassLoading = true
        deduceEnvironment = true
        optimizeNetty = true
        replaceLogbackXml = false
    }

    // Our API is too big to fit in a 32bit shadowJar zip
    shadowJar {
        zip64 true
    }
}
```

### Actual Behaviour

I get this error:
```
me@mylaptop myapp % ./gradlew optimizedJitJarAll
> Task :optimizedJitJarAll FAILED

[Incubating] Problems report is available at: file:///Users/me/git/myapp/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':optimizedJitJarAll'.
> org.apache.tools.zip.Zip64RequiredException: archive contains more than 65535 entries.
  
  To build this archive, please enable the zip64 extension.
  See: https://docs.gradle.org/8.12/dsl/org.gradle.api.tasks.bundling.Zip.html#org.gradle.api.tasks.bundling.Zip:zip64

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.12/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 18s
9 actionable tasks: 1 executed, 8 up-to-date
me@mylaptop myapp % 

```

### Steps To Reproduce

1. Have a large API
2. Configure gradle with `shadowJar { zip64 true }`
3. Run `./gradlew optimizedJitJarAll`
4. Still observe error

### Environment Information

openjdk 17.0.10 2024-01-16
OpenJDK Runtime Environment (build 17.0.10+0-17.0.10b1087.21-11572160)
OpenJDK 64-Bit Server VM (build 17.0.10+0-17.0.10b1087.21-11572160, mixed mode)
macOS Sequoia 15.3

### Example Application

_No response_

### Version

4.7.4

--- a/build.gradle
+++ b/build.gradle
@@ -38,9 +38,11 @@ dependencies {
     javadoc project(":micronaut-gradle-plugin")
     javadoc project(":micronaut-openapi-plugin")
     javadoc project(":micronaut-platform-catalog-plugin")
+    javadoc project(":micronaut-jsonschema-plugin")
     javadoc libs.micronaut.openapi.generator
     javadoc libs.bundles.optionalPlugins
     javadoc libs.shadowPlugin
+    javadoc libs.micronaut.jsonschema.generator
 }
 
 asciidoctorj {

--- a/src/docs/asciidoc/index.adoc
+++ b/src/docs/asciidoc/index.adoc
@@ -1379,6 +1379,26 @@ The task will generate a fat jar in the `build/libs` directory, that you can run
 
 `java -jar build/libs/myapp-0.1-all-optimized.jar`
 
+[NOTE]
+====
+Should you have configured your shadow jar specifically, for example to enable `zip64`, then you must also configure the `optimizedJitJarAll` task similarly.
+It is therefore recommended to use the `withType` method instead of specifically configuring the main shadow task:
+
+[source, groovy, subs="verbatim,attributes", role="multi-language-sample"]
+----
+tasks.withType(ShadowJar).configureEach {
+    zip64 = true
+}
+----
+
+[source, kotlin, subs="verbatim,attributes", role="multi-language-sample"]
+----
+tasks.withType<ShadowJar>().configureEach {
+    zip64.set(true)
+}
+----
+====
+
 [[aot:running-optimized-native-binary]]
 === Building and running an optimized native application
 
@@ -1408,7 +1428,6 @@ The 2 tasks which are available for this are:
 
 - `optimizedDockerBuildNative` to build the optimized native Docker image
 - `optimizedDockerPushNative` to push the optimized native Docker image
-
 
 [[test-resources]]
 == Automatic test resources provisioning

